packages/agoric-cli/src/lib/wallet.js

```diff
@@ -21,24 +21,23 @@ export const getWalletState = async (addr, ctx, { vstorage }) => {
   /** @type {import('./format').AssetDescriptor[]} */
   const brands = [];
   const mm = boardSlottingMarshaller(ctx.convertSlotToVal);
-  capDatas.forEach(capData => {
+  for (const capData of capDatas) {
     /** @type import('@agoric/smart-wallet/src/smartWallet').UpdateRecord */
     const update = mm.unserialize(capData);
     console.warn('wallet update', update);
     assert(update.updated, 'missing key: updated');
     switch (update.updated) {
       case 'offerStatus': {
         const { status } = update;
-        if (!offers.has(status.id)) {
-          offers.set(status.id, status);
-        }
+        const lastStatus = offers.get(status.id);
+        // merge records
+        offers.set(status.id, { ...lastStatus, ...status });
```

Comment:
I'd like a unit test for this fix. Does it look straightforward?

Reply:
If I'm going to put more work into this, I think I'd instead make it use [coalesceUpdates](https://github.com/Agoric/agoric-sdk/blob/f800d07a5c8c64791778a45a8db90d27c5181ad2/packages/smart-wallet/src/utils.js#L14) which does have some test coverage. That way other UIs could use the same lib.

Will that suffice? I can include that change in this PR.

Reply:
yes, that sounds good.